bdai_ros2_wrappers/test/test_subscription.py

```diff
@@ -22,3 +25,64 @@ def deferred_publish() -> None:
     message = wait_for_message(Int8, "test", node=ros.node, timeout_sec=10.0)
     assert message is not None
     assert message.data == 1
+
+
+def test_wait_for_messages(ros: ROSAwareScope) -> None:
+    class NodeFoo(Node):
+        def __init__(self, **kwargs: Any) -> None:
+            super().__init__("foo", **kwargs)
+            self.pub = self.create_publisher(std_msgs.msg.String, "/test1", 10)
+            self.timer = self.create_timer(0.5, self.publish_msg)
+
+        def publish_msg(self) -> None:
+            self.pub.publish(std_msgs.msg.String(data="hello from foo"))
+
+    class NodeBar(Node):
```

Comment:
@kaiyu-zheng nit^2: you could reuse the same node class and make it configurable (hint: you can `ros.load(NodeClass, arg1, arg2, kwarg1=..., kwarg2=...)`)